### PR TITLE
Enhancing Model Accuracy through Advanced Transfer Learning Techniques

### PR DESCRIPTION
**Pull request by DarshAgrawal14**
Issue : #8 
-The current implementation utilizes EfficientNet for terrain classification.
- Data Agumentation  includes resizing, horizontal flipping, color jitter, and random rotations.
- I have loaded a pre-trained EfficientNet B0 model and modified the classifier layer to match the number of output classes (5 for different terrain types).
-  I noticed at high learning rates the accuracy was highly volatile so it was lowered
- I defined the loss function (CrossEntropyLoss) and optimizer (Adam) with a lower learning rate for fine-tuning.
- Learning rate scheduling : ReduceLROnPlateau
- Included code to load model and test it using a example image
- The hyper - parameter tuning was performed using gpu , but if not available code will automatically use cpu

Darsh Agrawal , GSSoC 2024 extd contributor

### CODE REVIEW

**Comment by Akasxh** [comment]
I'll merge once I've reviewed it.

**Comment by DarshAgrawal14** [comment]
Sure!
